### PR TITLE
updated logging, mongo config

### PR DESCRIPTION
**Pull request by andricDu**
Properly configured Winston logging with colourized output

image

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -38,6 +39,8 @@ app.use(bodyParser.urlencoded({
 swagger["info"]["version"] = process.env.npm_package_version;
 app.use("/api-docs", swaggerUi.serve, swaggerUi.setup(swagger));
 
+logger.info("Swagger setup... done");
```
Comment (joneubank) [inline src/app.ts]: nice to have: console out the URL for swagger

File: src/server.ts [hunk 2]
```diff
@@ -16,12 +47,10 @@ mongoose.connect(constructMongoUri(), { useNewUrlParser: true }).then(
  * Start Express server.
  */
 const server = app.listen(app.get("port"), () => {
-  console.log(
-    "  App is running at http://localhost:%d in %s mode",
-    app.get("port"),
-    app.get("env")
+  logger.info(
+    `App is running at http://localhost:${app.get("port")} in ${app.get("env")} mode`
```
Comment (joneubank) [inline src/server.ts]: 👍

File: src/utils/errors.ts [hunk 3]
```diff
@@ -78,6 +79,10 @@ export const errorHandler = (err: Error, req: Request, res: Response, next: Next
             res.status(400);
             break;
         default:
+            logger.error("Internal Server Error");
+            logger.error(err.name);
+            logger.error(err.message);
+            logger.error(err.stack);
```
Comment (joneubank) [inline src/utils/errors.ts]: These four messages could be separated by logs from other requests happening at same time, even another error causing debug confusion... perhaps you want these 4 messages combined into one?

**Comment by andricDu** [comment]
Thanks @joneubank

**Pull request merged**